R/deps.R

```diff
@@ -47,9 +44,6 @@
 #' }
 package_deps <- function(pkg, dependencies = NA, repos = getOption("repos"),
                          type = getOption("pkgType")) {
-  if (identical(type, "both")) {
```

Comment:
devtools always forces a type of `binary` https://github.com/hadley/devtools/pull/1339/files#diff-e7118181d76027ea30ccf5babc54f550L51, which means you never have a interactive dialog asking if you want to install.

This PR sets `option("install.packages.compile.from.source" = "never")`, which effectively means the same as forcing a type of "binary" if type = 'both', _unless_ there is no binary available (such as drat repositories), in which case it will compile the package.